de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/simulator/JSSimulatorNodeModel.java

```diff
@@ -96,28 +92,36 @@ public ValidationError validateViewValue(JSSimulatorViewValue viewContent) {
   }
 
   @Override
-  public void saveCurrentValue(NodeSettingsWO content) {}
+  public void saveCurrentValue(NodeSettingsWO content) {
+  }
 
   @Override
   public JSSimulatorViewValue getViewValue() {
-    JSSimulatorViewValue val;
+
+    JSSimulatorViewValue value = super.getViewValue();
     synchronized (getLock()) {

```

Comment:
In getViewValue, the simulations must be taken first from settings (if configured) and then from port object as a fall back.

```java
      // Copy simulation from settings
      if (value.getSimulations() == null || value.getSimulations().length == 0) {
        copyConfigToView(value);
      }

      // If still not initialized, then copy from input model
      if (value.getSimulations() == null || value.getSimulations().length == 0) {
        // Convert from FskSimulation(s) to JSSimulation(s)
        value.setSimulations(convertSimulations(port.modelMetadata));
        value.setSelectedSimulationIndex(port.selectedSimulationIndex);
      }
```